dotnet/src/Plugins/Plugins.Web/Bing/BingTextSearchOptions.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Search;
+
+namespace Microsoft.SemanticKernel.Plugins.Web.Bing;
+
+/// <summary>
+/// Options used to construct an instance of <see cref="BingTextSearch"/>
+/// </summary>
+public sealed class BingTextSearchOptions
+{
+    /// <summary>
+    /// The URI endpoint of the Bing search service. The URI must use HTTPS.
+    /// </summary>
+    public Uri? Endpoint { get; init; } = null;
+
+    /// <summary>
+    /// The HTTP client to use for making requests.
+    /// </summary>
+    public HttpClient? HttpClient { get; init; } = null;
+
+    /// <summary>
+    /// The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.
+    /// </summary>
+    public ILoggerFactory? LoggerFactory { get; init; } = null;
+
+    /// <summary>
+    ///  Delegate to map a <see cref="BingWebPage"/> instance to a <see cref="string"/>
+    /// </summary>
+    public MapBingWebPageToString? MapToString { get; init; } = null;
```

Comment:
I wonder whether we should use nouns for the property and the one below because verbs are reserved for methods. `MapToString` -> `ResultToStringMapper`? and `MapToTextSearchResult` -> `ResultToTextSearchResultMapper` or something along these lines.

Reply:
I will be changing these to be interfaces to match the pattern used in the Vector Store, see #8390